frontend/schema/class-schema-article.php

```diff
@@ -61,10 +62,13 @@ public function generate() {
 			'headline'         => get_the_title(),
 			'datePublished'    => mysql2date( DATE_W3C, $post->post_date_gmt, false ),
 			'dateModified'     => mysql2date( DATE_W3C, $post->post_modified_gmt, false ),
-			'commentCount'     => $comment_count['approved'],
 			'mainEntityOfPage' => array( '@id' => $this->context->canonical . WPSEO_Schema_IDs::WEBPAGE_HASH ),
 		);
 
+		if ( $comments_open || $comment_count['approved'] !== 0 ){
```

Comment:
Would this be better like this:

`if ( $comment_count['approved'] > 0 ) {`

Seems easier to me.

Reply:
We can go for that as well, in that case we don't need to check for comments-disabled. But, why not show the comment count as 0 when comments are enabled?